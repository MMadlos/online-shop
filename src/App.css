/* VARIABLES */
:root {
	/* COLORS */
	--black: #333333;
	--white: #fafafa;
	--accent-purple: #7443ff;
	--light-purple: #e3d9ff;
	--accent-yellow: #e2fe91;
	--yellow-star: #ffda17;
	--light-grey: #f2f2f2;
	--grey: #e5e6ea;
	--dark-grey: #c7c9d1;
	--light-black: #72778d;
	--alert: #fa7373;
	--light-green: #e3fee4;
	--green: #15a409;

	/* OTHERS */
	--padding-section-laterals: 15vw;
	--padding-media-small: 16px;
	--font-main: "Inter", sans-serif;
	--radial-topleft-purple: radial-gradient(circle at top left, var(--light-purple) 25%, var(--white) 100%);
	--radial-topright-purple: radial-gradient(circle at top right, var(--light-purple) 25%, var(--white) 100%);
}

/* GENERIC */

body {
	font-family: var(--font-main);
}

#root {
	/* Added to grow the main content when it doesn't fill the available heigh*/
	height: 100vh;

	display: flex;
	flex-direction: column;
}

footer {
	background-color: var(--black);
	color: var(--white);
	padding: 20px 10vw;
}

h1 {
	font-size: 1.5rem;
	font-weight: 900;
}

h2 {
	font-size: 1.25rem;
	font-weight: 800;
}

ul {
	list-style-type: none;
}

a {
	text-decoration-line: none;
	color: var(--black);
}

button {
	outline: none;
	border: none;
}

@media (max-width: 500px) {
	h1 {
		font-size: 16px;
	}
}

/* ACTION BAR */
.action-bar {
	display: flex;
	justify-content: space-between;

	background-color: var(--white);
	padding: 16px var(--padding-section-laterals);
	padding-bottom: 0px;

	> form {
		display: flex;
		align-items: center;
		gap: 16px;

		padding: 4px 16px;
		background-color: var(--white);

		border: 1px solid var(--dark-grey);
		border-radius: 4px;

		width: 80%;
		max-width: 600px;

		> i {
			color: var(--dark-grey);
		}

		> input {
			border: none;
			background-color: var(--white);
			color: var(--black);
			font-size: 14px;
			width: 100%;

			&::placeholder {
				color: var(--dark-grey);
			}

			&:focus {
				outline: none;
			}

			&::-webkit-search-cancel-button {
				cursor: pointer;
			}
		}
	}

	#search-not-found {
		color: red;
	}
}

@media (max-width: 500px) {
	.action-bar {
		flex-direction: column;
		gap: 8px;
		padding: 12px var(--padding-media-small);

		> form {
			gap: 16px;
			padding: 8px;
			width: 100%;
		}
	}
}

.dropdown-container {
	position: relative;

	> button#sort {
		display: flex;
		justify-content: space-between;
		align-items: center;
		gap: 32px;
		padding: 8px 16px;

		background-color: var(--grey);
		border: 1px solid var(--dark-grey);
		border-radius: 2px;

		color: var(--light-black);
		font-weight: 600;

		&:hover,
		&.selected {
			background-color: var(--light-black);
			color: var(--white);
		}

		> i {
			font-size: 1rem;
		}
	}

	.dropdown-menu {
		position: absolute;
		transform-origin: top right;
		right: 0;
		width: 250px;

		background-color: var(--white);

		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: start;

		border: 1px solid var(--dark-grey);
		border-radius: 2px;

		padding: 8px;
		margin-top: 8px;

		.item {
			color: var(--black);
			width: 100%;
			padding: 8px 16px;
			text-align: left;
			border-radius: 2px;
			background-color: transparent;

			&:hover {
				background-color: var(--light-grey);
			}
		}

		.selected {
			background-color: var(--grey);
			color: var(--black);
			font-weight: 600;
		}

		.group-title {
			font-size: 12px;
			font-weight: 700;
			color: var(--light-black);
		}

		.divider {
			height: 1px;
			background-color: var(--grey);
			width: 100%;
			margin: 8px 0;
		}
	}
}

@media (max-width: 500px) {
	.dropdown-container {
		> button#sort {
			gap: 8px;
			padding: 8px;
			width: 100%;
			font-weight: 500;

			> i {
				font-size: 0.75rem;
			}
		}

		.dropdown-menu {
			transform-origin: top left;
			left: 0;
			width: 100%;
		}
	}
}

.action-messages {
	display: flex;
	justify-content: space-between;
	padding: 0 var(--padding-section-laterals);
	height: 50px;
	background-color: var(--white);

	.search-not-found {
		color: var(--alert);
		font-size: 14px;
		font-weight: 500;
		width: 100%;
	}

	.chip-sort {
		color: var(--black);
		font-size: 12px;
		width: 100%;
		text-align: right;

		display: flex;
		justify-content: end;
		align-items: center;
		gap: 12px;
		padding-top: 16px;

		.sort-selected {
			display: flex;
			justify-content: space-between;
			align-items: center;
			gap: 12px;

			background-color: var(--grey);
			padding: 4px 16px;
			border: 1px solid var(--dark-grey);
			border-radius: 50px;

			> i {
				color: var(--light-black);

				&:hover {
					color: var(--dark-grey);
				}
			}
		}
	}
}

@media (max-width: 500px) {
	.action-messages {
		flex-direction: column;
		padding: 0 var(--padding-media-small);
		gap: 8px;
		height: fit-content;

		.search-not-found {
			font-size: 12px;
			font-weight: 700;
		}

		.chip-sort {
			font-size: 12px;
			width: 100%;

			justify-content: flex-start;
			padding-top: 0;
			margin-bottom: 8px;

			.sort-selected {
				gap: 12px;

				padding: 4px 12px;

				> i {
					font-size: 1rem;
					color: var(--light-black);
				}
			}
		}
	}
}

/* PRODUCT INFO */
section.product-info {
	padding: 16px var(--padding-section-laterals);
	color: var(--black);

	> a {
		display: flex;
		align-items: center;
		gap: 8px;

		font-size: 14px;
		font-weight: 500;
		color: var(--light-black);

		&:hover {
			color: var(--black);
		}
	}

	.product-details-container {
		padding-top: 40px;
		display: flex;
		gap: 80px;

		.image-container {
			min-width: 40%;
			max-height: 65vh;
			max-width: 60%;

			> img {
				object-fit: contain;
				object-position: center;
				height: 100%;
				width: 100%;
			}
		}

		.product-details {
			display: flex;
			flex-direction: column;
			justify-content: space-between;

			background-color: var(--light-grey);
			padding: 40px;
			border-radius: 8px;
			min-width: 300px;
			min-height: 50vh;

			.details-container {
				display: flex;
				flex-direction: column;
				gap: 16px;

				> h1 {
					font-size: 2rem;
					font-weight: 800;
				}

				.rate-container {
					display: flex;
					align-items: center;
					gap: 8px;
					color: var(--light-black);
					font-size: 14px;
					font-weight: 600;
					> i {
						color: var(--yellow-star);
					}
				}

				.description {
					font-size: 14px;
					line-height: 150%;
					font-weight: 500;
					letter-spacing: 0.25px;
					color: var(--light-black);
				}

				.price {
					font-size: 24px;
					font-weight: 700;
					color: var(--light-black);
				}
			}

			.add-cart-container {
				display: flex;
				justify-content: space-between;
				align-items: center;
				flex-wrap: wrap;
				gap: 40px;
				row-gap: 16px;

				.counter {
					min-width: fit-content;
					max-width: 100%;

					display: flex;
					align-items: center;
					gap: 8px;

					#increase,
					#decrease {
						font-size: 14px;
						color: var(--black);
						background-color: var(--grey);
						padding: 8px;
						width: 35px;
						aspect-ratio: 1 / 1;

						border-radius: 4px;

						&:hover:not(:disabled) {
							color: var(--black);
							background-color: var(--dark-grey);
						}

						&:disabled {
							opacity: 50%;

							> i {
								opacity: 25%;
							}
						}
					}

					.quantity {
						display: flex;
						align-items: center;
						justify-content: center;
						padding: 8px 24px;
						font-weight: 600;
						background-color: var(--white);
						border-radius: 4px;
						width: 100%;
					}
				}

				> button#add-cart {
					min-width: fit-content;
					max-width: 100%;
					flex-grow: 1;
					font-weight: 700;
					font-size: 14px;
					padding: 12px 24px;

					border-radius: 2px;
					color: var(--white);
					background-color: var(--black);

					&:hover {
						cursor: pointer;
						background-color: var(--light-black);
					}
				}
			}
		}
	}
}

@media (max-width: 500px) {
	section.product-info {
		padding: 16px var(--padding-media-small);
		color: var(--black);

		.product-details-container {
			padding-top: 16px;
			flex-direction: column;
			align-items: center;
			gap: 24px;

			.image-container {
				height: 150px;

				> img {
					object-fit: contain;
					object-position: center;
				}
			}

			.product-details {
				padding: var(--padding-media-small);
				gap: 16px;
				min-height: fit-content;

				.details-container {
					display: flex;
					flex-direction: column;
					gap: 12px;

					> h1 {
						font-size: 1.25rem;
						font-weight: 700;
					}

					.rate-container {
						font-size: 12px;
						font-weight: 500;
					}

					.description {
						line-height: 125%;
						font-weight: 500;
						letter-spacing: 0.15px;
					}

					.price {
						font-size: 1rem;
					}
				}

				.add-cart-container {
					.counter {
						min-width: fit-content;
						width: 100%;

						display: flex;
						align-items: center;
						gap: 8px;

						#increase,
						#decrease {
							font-size: 12px;
							padding: 8px 24px;
							/* width: 35px; */
							aspect-ratio: 1/1;

							border-radius: 50px;

							> i {
								display: flex;
								justify-content: center;
								color: var(--light-black);
							}
						}

						.quantity {
							display: flex;
							align-items: center;
							justify-content: center;
							padding: 8px 24px;
							font-weight: 600;
							border-radius: 4px;
							width: 100%;
						}
					}
				}

				/* .quantity {
					font-size: 14px;
				} */
			}
		}
	}
}

/* SECTION - CART */
section#cart {
	padding: 16px var(--padding-section-laterals) 40px var(--padding-section-laterals);

	display: flex;
	flex-direction: column;
	gap: 8px;
	color: var(--black);

	.return-container {
		display: flex;
		gap: 8px;
		padding: 8px 8px 8px 0;
		align-items: center;
		font-weight: 500;

		&:hover {
			cursor: pointer;
			opacity: 50%;
		}
	}

	> h1 {
		margin-top: 24px;
		font-weight: 700;
	}

	.product-card.cart {
		background-color: var(--white);
		border: 1px solid var(--dark-grey);
		border-radius: 8px;
		padding: 8px;

		&:hover {
			border-color: var(--light-black);
		}

		.product-info {
			display: flex;
			justify-content: space-between;
			align-items: center;
			gap: 32px;

			padding: 8px 16px;

			.product-info-details-container {
				display: flex;
				gap: 24px;
				max-width: 60%;

				> img {
					--side: 100px;
					max-height: var(--side);
					max-width: var(--side);
					aspect-ratio: 1 / 1;

					object-position: center;
					object-fit: contain;
				}

				.produt-info-name-price-container {
					display: flex;
					flex-direction: column;
					gap: 8px;

					.name {
						font-size: 16px;
						font-weight: 700;
					}

					.description {
						font-size: 14px;
						color: var(--light-black);

						-webkit-line-clamp: 2;
						-webkit-box-orient: vertical;
						overflow: hidden;
						text-overflow: ellipsis;
						display: -webkit-box;
					}

					.price {
						font-size: 1.25rem;
						font-weight: 600;
					}
				}
			}

			.counter {
				min-width: fit-content;
				max-width: 100%;

				display: flex;
				align-items: center;
				gap: 8px;

				#increase,
				#decrease {
					font-size: 12px;
					color: var(--black);
					background-color: var(--grey);
					padding: 8px;
					width: 35px;
					aspect-ratio: 1 / 1;

					border-radius: 4px;

					&:hover:not(:disabled) {
						color: var(--black);
						background-color: var(--dark-grey);
					}

					&:disabled {
						opacity: 50%;

						> i {
							opacity: 25%;
						}
					}
				}

				.quantity {
					display: flex;
					align-items: center;
					justify-content: center;
					padding: 8px 24px;
					font-weight: 600;
					border-radius: 4px;
					width: 100%;
				}
			}

			#remove-cart {
				font-weight: 700;
				font-size: 14px;
				padding: 8px 24px;
				width: fit-content;
				border-radius: 2px;

				color: var(--light-black);
				background-color: var(--grey);
				font-weight: 500;

				&:hover {
					cursor: pointer;
					background-color: var(--dark-grey);
				}
			}
		}
	}

	.check-out-container {
		display: flex;
		justify-content: space-between;
		align-items: center;

		gap: 16px;
		padding: 16px 16px;

		border-radius: 6px;
		background-color: var(--light-grey);

		.total-container > p {
			font-size: 1.25rem;

			> span {
				padding-left: 16px;
				font-weight: 700;
			}
		}

		> button#checkout {
			font-size: 16px;
			font-weight: 700;
			padding: 8px 32px;
			border-radius: 2px;
			letter-spacing: 0.25px;

			color: var(--white);
			background-color: var(--accent-purple);

			&:hover {
				cursor: pointer;
				opacity: 50%;
			}
		}
	}

	.cart-emtpy {
		display: flex;
		justify-content: center;
		padding: 40px 0;
		border: 4px;
		color: var(--light-black);
		background-color: var(--light-grey);
	}
}

@media (max-width: 500px) {
	section#cart {
		padding: 16px var(--padding-media-small);

		display: flex;
		flex-direction: column;
		gap: 8px;
		color: var(--black);

		.return-container {
			padding: 0;
			font-size: 14px;
			color: var(--light-black);

			> a {
				font-weight: 500;
				color: var(--light-black);
			}
		}

		> h1 {
			margin-top: 8px;
			font-weight: 700;
		}

		> a {
			.product-card.cart {
				.product-info {
					/* display: flex; */
					flex-direction: column;
					/* justify-content: space-between; */
					/* align-items: center; */
					gap: 16px;

					/* padding: 8px 16px; */

					.product-info-details-container {
						/* display: flex; */
						gap: 24px;
						max-width: 100%;

						flex-direction: column;

						> img {
							max-height: 20vh;
							max-width: 100%;

							object-position: center;
							object-fit: contain;
						}

						.produt-info-name-price-container {
							display: flex;
							flex-direction: column;
							gap: 8px;

							.name {
								font-size: 16px;
								font-weight: 700;

								-webkit-line-clamp: 2;
								-webkit-box-orient: vertical;
								overflow: hidden;
								text-overflow: ellipsis;
								display: -webkit-box;
							}

							.description {
								display: none;
							}

							.price {
								font-size: 1rem;
								font-weight: 500;
								color: var(--light-black);
							}
						}
					}

					.counter {
						min-width: fit-content;
						width: 100%;

						display: flex;
						align-items: center;
						gap: 8px;

						#increase,
						#decrease {
							font-size: 12px;
							padding: 8px 24px;
							/* width: 35px; */
							aspect-ratio: 1/1;

							border-radius: 50px;

							> i {
								display: flex;
								justify-content: center;
								color: var(--light-black);
							}
						}

						.quantity {
							display: flex;
							align-items: center;
							justify-content: center;
							padding: 8px 24px;
							font-weight: 600;
							border-radius: 4px;
							width: 100%;
						}
					}

					#remove-cart {
						width: 100%;
						border-radius: 4px;
					}
				}
			}
		}

		.check-out-container {
			flex-direction: column;
			padding: 16px 8px;

			> button#checkout {
				width: 100%;
			}
		}
	}
}

/* RECOMMENDATION SECTION */

section.recommendation {
	--padding-top-bottom: 40px;

	padding: var(--padding-top-bottom) var(--padding-section-laterals);
	background-color: var(--grey);

	> h2 {
		font-size: 1.5rem;
		font-weight: 600;
		color: var(--accent-purple);
		margin-bottom: 40px;
	}

	.recommendation-container {
		display: grid;
		grid-template-columns: repeat(4, 1fr);
		align-items: center;
		gap: 16px;

		> a {
			height: 100%;

			.product-card {
				background-color: var(--light-grey);
				width: 100%;

				overflow: hidden;
				display: flex;
				flex-direction: column;
				height: 100%;

				border: 1px solid var(--dark-grey);
				border-radius: 4px;

				&:hover {
					border: 1px solid var(--black);
				}

				> img {
					min-height: 150px;
					max-height: 250px;
					height: 100%;
					object-fit: contain;
					object-position: center;
					padding: 32px 64px;
					background-color: var(--white);
				}

				.product-info {
					display: flex;
					justify-content: space-between;

					flex-direction: column;
					gap: 16px;
					background-color: var(--light-grey);
					padding: 16px;

					.product-info-container {
						display: flex;
						flex-direction: column;
						gap: 12px;

						#name {
							-webkit-line-clamp: 2;
							-webkit-box-orient: vertical;
							overflow: hidden;
							text-overflow: ellipsis;
							display: -webkit-box;
							font-size: 1rem;
							font-weight: 600;
							line-height: 130%;
						}

						.rate-container {
							display: flex;
							align-items: center;
							margin-top: -8px;
							gap: 8px;
							font-size: 12px;
							color: var(--light-black);

							.fa-star {
								color: var(--yellow-star);
							}
						}

						#price {
							font-size: 14px;
							font-weight: 600;
							color: var(--light-black);
						}
					}
				}
			}
		}
	}
}

@media (max-width: 500px) {
	section.recommendation {
		padding: var(--padding-top-bottom) var(--padding-media-small);
		background-color: var(--white);

		> h2 {
			font-size: 1.25rem;
			font-weight: 600;
			margin-bottom: 16px;
		}

		.recommendation-container {
			display: grid;
			grid-template-columns: repeat(4, 1fr);
			align-items: center;
			gap: 8px;

			overflow-x: scroll;

			> a {
				height: 100%;

				.product-card {
					background-color: var(--light-grey);
					min-width: 200px;

					border-radius: 8px;

					&:hover {
						border: 1px solid var(--black);
					}

					> img {
						height: 200px;
						width: 100%;

						padding: 16px;
					}
				}
			}
		}
	}
}

/* .home {
	background-color: var(--accent-yellow);

	> main {
		background-color: transparent;
	}
} */
