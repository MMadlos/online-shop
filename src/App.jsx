import "./App.css"
import { Outlet } from "react-router-dom"
import { createContext, useEffect, useState } from "react"
import { filterProductsByCategory, getProductByID, toggleIsProductAddedTo, replaceProductInList, getCategories } from "./utilities/utilities"

import useFetchProducts from "./hooks/useFetchProducts"
import useDisplayAndSort from "./hooks/useDisplayndSort"

import Header from "./components/Header"

export const ShopContext = createContext({
	error: null,
	loading: true,
	productList: [],
	setProductList: () => {},
	selectedCategory: "Men's clothing",
	cartList: [],
	setCartList: () => {},
	productsToShow: [],
	categoryList: [],

	handleClickCategory: () => {},
	handleClickAddCart: () => {},
	handleClickRemoveCart: () => {},
})

export const SearchAndSortContext = createContext({
	sort: {},
	isProductFound: true,

	setSort: () => {},
	handleClickSort: () => {},
	handleChangeSearch: () => {},
})

function App() {
	const { productList, setProductList, error, loading } = useFetchProducts()

	const categoryList = ["All categories", ...getCategories(productList)]
	const [selectedCategory, setSelectedCategory] = useState("Men's clothing")
	const { sort, setSort, productsToShow, setProductsToShow } = useDisplayAndSort()

	const [cartList, setCartList] = useState([])
	const [isProductFound, setIsProductFound] = useState(true)

	useEffect(() => {
		const filteredProducts = filterProductsByCategory(productList, selectedCategory)
		setProductsToShow(filteredProducts)
	}, [productList, selectedCategory])

	const handleClickCategory = (e) => setSelectedCategory(e.target.textContent)

	function handleClickAddCart(e) {
		e.preventDefault()
		e.stopPropagation()

		const getSection = e.target.closest(".product-card") || e.target.closest(".product-details-container")

		const getClass = getSection.classList.value
		const { dataset } = getSection

		const productID = Number(dataset.productId)
		const productSelected = getProductByID(productsToShow, productID)

		if (getClass === "product-card") {
			setCartList((prev) => [...prev, productSelected])

			productSelected.quantity++
			const newProductList = toggleIsProductAddedTo(true, productSelected, productList)
			setProductList(newProductList)
		}
		if (getClass === "product-details-container") {
			const containerElement = e.target.closest(".add-cart-container")
			const counterElement = containerElement.querySelector(".counter > p")
			const counterQuantity = Number(counterElement.textContent)

			productSelected.quantity += counterQuantity

			const isProductInCart = cartList.some((product) => product.id === productSelected.id)

			if (isProductInCart) {
				const newCartList = replaceProductInList(productSelected, cartList)
				setCartList(newCartList)

				const newProductList = replaceProductInList(productSelected, productList)
				setProductList(newProductList)
				return
			}
			if (!isProductInCart) {
				setCartList((prev) => [...prev, productSelected])

				const newProductList = toggleIsProductAddedTo(true, productSelected, productList)
				setProductList(newProductList)
				return
			}
		}
	}

	function handleClickRemoveCart(e) {
		e.preventDefault()
		e.stopPropagation()

		const { dataset } = e.target.closest("[data-product-id]")
		const productID = Number(dataset.productId)

		const newCart = cartList.filter((product) => product.id !== productID)
		setCartList(newCart)

		const productSelected = getProductByID(productList, productID)
		productSelected.quantity = 0

		const newProductList = toggleIsProductAddedTo(false, productSelected, productList)
		setProductList(newProductList)
	}

	function handleChangeSearch(e) {
		const query = e.target.value

		const filteredProducts = filterProductsByCategory(productList, selectedCategory)
		const productsFound = filteredProducts.filter(
			(product) => product.name.toLowerCase().includes(query.toLowerCase()) || product.name.startsWith(query.toLowerCase())
		)

		setIsProductFound(productsFound.length === 0 ? false : true)
		setProductsToShow(productsFound.length === 0 ? filteredProducts : productsFound)
	}

	function handleClickSort(e) {
		const { groupName } = e.target.dataset
		const itemName = e.target.textContent
		const selectedItem = { group: groupName, item: itemName }
		setSort(selectedItem)
	}

	return (
		<>
			<Header cartQuantity={cartList.length} />

			<ShopContext.Provider
				value={{
					error,
					loading,
					productList,
					categoryList,
					selectedCategory,
					cartList,
					productsToShow,
					setProductList,
					setCartList,

					handleClickCategory,
					handleClickAddCart,
					handleClickRemoveCart,
				}}>
				<SearchAndSortContext.Provider value={{ sort, isProductFound, setSort, handleChangeSearch, handleClickSort }}>
					<Outlet />
				</SearchAndSortContext.Provider>
			</ShopContext.Provider>
			{/* <footer>
				<p>Created by M.Madlos</p>
			</footer> */}
		</>
	)
}

export default App
