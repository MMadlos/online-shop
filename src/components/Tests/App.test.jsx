import { render, screen } from "@testing-library/react"
import { describe, it, expect } from "vitest"
import userEvent from "@testing-library/user-event"

import App from "../../App"

describe("App", () => {
	it("Renders App", () => {
		const { container } = render(<App />)

		expect(container).toMatchSnapshot()
	})

	it("renders without any item in the cart", () => {
		render(<App />)
		expect(screen.getByText("Cart (0)")).toBeInTheDocument()
	})

	it("Increments or decrements cart number when clicking in add to cart button", async () => {
		const user = userEvent.setup()

		render(<App />)

		const allButtons = screen.getAllByText("Add to cart")
		const addCartBtn = allButtons[0]

		await user.click(addCartBtn)

		expect(screen.getByText("Cart (1)")).toBeInTheDocument()

		const removeBtn = screen.getByText("Remove from cart")
		await user.click(removeBtn)

		expect(screen.getByText("Cart (0)")).toBeInTheDocument()
	})

	it("initially selects 'Men's clothing' category", () => {
		const { container } = render(<App />)

		const categorySelected = container.querySelector(".selected")
		expect(categorySelected.textContent).toBe("Men's clothing")
	})
})
