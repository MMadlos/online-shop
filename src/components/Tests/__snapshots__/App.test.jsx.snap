// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Renders App 1`] = `
<div>
  <header>
    <div
      class="logo-container"
    >
      <h1>
        Online shop
      </h1>
    </div>
    <ul>
      <li>
        <a
          href="#"
        >
          Ver en Github
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Cart (
          0
          )
        </a>
      </li>
    </ul>
  </header>
  <div
    class="category-container"
  >
    <div
      class="category"
    >
      <a>
        All categories
      </a>
    </div>
    <div
      class="category selected"
    >
      <a>
        Men's clothing
      </a>
    </div>
    <div
      class="category"
    >
      <a>
        Jewelery
      </a>
    </div>
    <div
      class="category"
    >
      <a>
        Electronics
      </a>
    </div>
    <div
      class="category"
    >
      <a>
        Women's clothing
      </a>
    </div>
  </div>
  <main>
    <div
      class="product-card"
      data-product-id="1"
    >
      <img
        alt="#"
        src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
      />
      <div
        class="product-info"
      >
        <div
          class="product-info-container"
        >
          <h2
            id="name"
          >
            Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
          </h2>
          <p
            id="description"
          >
            Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
          </p>
          <p
            id="price"
          >
            $ 109.95
          </p>
          <div
            class="rate-container"
          >
            <p
              id="rate"
            >
              3.9
            </p>
            <i
              class="fa-solid fa-star-half-stroke"
            />
            <p
              id="count-rates"
            >
              120
            </p>
          </div>
        </div>
        <button
          id="add-cart"
        >
          Add to cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
      data-product-id="2"
    >
      <img
        alt="#"
        src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
      />
      <div
        class="product-info"
      >
        <div
          class="product-info-container"
        >
          <h2
            id="name"
          >
            Mens Casual Premium Slim Fit T-Shirts 
          </h2>
          <p
            id="description"
          >
            Slim-fitting style, contrast raglan long sleeve, three-button henley placket, light weight & soft fabric for breathable and comfortable wearing. And Solid stitched shirts with round neck made for durability and a great fit for casual fashion wear and diehard baseball fans. The Henley style round neckline includes a three-button placket.
          </p>
          <p
            id="price"
          >
            $ 22.3
          </p>
          <div
            class="rate-container"
          >
            <p
              id="rate"
            >
              4.1
            </p>
            <i
              class="fa-solid fa-star-half-stroke"
            />
            <p
              id="count-rates"
            >
              259
            </p>
          </div>
        </div>
        <button
          id="add-cart"
        >
          Add to cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
      data-product-id="3"
    >
      <img
        alt="#"
        src="https://fakestoreapi.com/img/71li-ujtlUL._AC_UX679_.jpg"
      />
      <div
        class="product-info"
      >
        <div
          class="product-info-container"
        >
          <h2
            id="name"
          >
            Mens Cotton Jacket
          </h2>
          <p
            id="description"
          >
            great outerwear jackets for Spring/Autumn/Winter, suitable for many occasions, such as working, hiking, camping, mountain/rock climbing, cycling, traveling or other outdoors. Good gift choice for you or your family member. A warm hearted love to Father, husband or son in this thanksgiving or Christmas Day.
          </p>
          <p
            id="price"
          >
            $ 55.99
          </p>
          <div
            class="rate-container"
          >
            <p
              id="rate"
            >
              4.7
            </p>
            <i
              class="fa-solid fa-star-half-stroke"
            />
            <p
              id="count-rates"
            >
              500
            </p>
          </div>
        </div>
        <button
          id="add-cart"
        >
          Add to cart
        </button>
      </div>
    </div>
    <div
      class="product-card"
      data-product-id="4"
    >
      <img
        alt="#"
        src="https://fakestoreapi.com/img/71YXzeOuslL._AC_UY879_.jpg"
      />
      <div
        class="product-info"
      >
        <div
          class="product-info-container"
        >
          <h2
            id="name"
          >
            Mens Casual Slim Fit
          </h2>
          <p
            id="description"
          >
            The color could be slightly different between on the screen and in practice. / Please note that body builds vary by person, therefore, detailed size information should be reviewed below on the product description.
          </p>
          <p
            id="price"
          >
            $ 15.99
          </p>
          <div
            class="rate-container"
          >
            <p
              id="rate"
            >
              2.1
            </p>
            <i
              class="fa-solid fa-star-half-stroke"
            />
            <p
              id="count-rates"
            >
              430
            </p>
          </div>
        </div>
        <button
          id="add-cart"
        >
          Add to cart
        </button>
      </div>
    </div>
  </main>
</div>
`;
